saas_apps/models/saas_apps.py

```diff
@@ -0,0 +1,150 @@
+                        new = self.create({
+                            'name': app.name,
+                            'module_name': ir_module_obj['name'],
+                            'icon_path': ir_module_obj['icon']
+                        })
+                        new.dependencies += apps.search([('name', '=', app.name)])
+                        for dep_name in ir_module_obj['depends']:
+                            new.dependencies += apps.search([('name', '=', dep_name)])
+                else:
+                    new = self.create({
+                        'name': app.name,
+                        'module_name': app.name,
+                        'icon_path': base_icon_path
+                    })
+
+    def _compute_year_price(self):
+        self.year_price = 0
+        for module in self.dependencies:
+            self.year_price += module.year_price
+
+    def _compute_month_price(self):
+        self.month_price = 0
+        for module in self.dependencies:
+            self.month_price += module.month_price
+
+    def dependencies_info(self, root):
+        apps = []
+        childs = []
+        saas_module = self.dependencies.search([('name', '=', self.name)])
+        for child in self.dependencies - saas_module:
+            if len(self.search([('name', '=', child.name)])):
+                childs.append(child.name)
+        apps.append({
+            'parent': root,
+            'name': self.name,
+            'year_price': saas_module.year_price,
+            'month_price': saas_module.month_price,
+            'childs': childs
+        })
+        # Looking to the period
+        for app in self.dependencies - saas_module:
+            set = self.search([('name', '=', app.name)])
+            if len(set) == 0:
+                continue
+            leafs = set.dependencies_info(self.name)
+            for leaf in leafs:
+                if not(leaf in apps):
+                    apps.append(leaf)
+        return apps
+
+    @api.multi
+    def write(self, vals):
+        res = super(SAASDependence, self).write(vals)
+        # If value of allow_to_sell changed, other sets allow_to_sell vars should be changed too
+        if "allow_to_sell" in vals:
+            this_app = self.dependencies.search([('name', '=', self.name)])
+            for app in self.dependencies - this_app:
+                temp_app = self.search([('name', '=', app.name)])
+                if len(temp_app) > 0:
+                    temp_app.allow_to_sell = vals['allow_to_sell']
+        return res
+
+
+class SAASTemplateLine(models.Model):
+    _inherit = 'saas.template.operator'
+
+
+    @api.multi
+    def write(self, vals):
+        if 'state' in vals:
+            _logger.debug(vals['state'])
+        return super(SAASTemplateLine, self).write(vals)
+
+    @api.multi
+    def random_ready_operator_check(self):
+        ready_operators = self.filtered(lambda r: r.state == 'done')
+        if len(ready_operators) > 0:
```

Comment:
I'm wondering, is it workable?
```
return bool(ready_operators)
```

Reply:
@KolushovAlexandr i wondered too, so i decided not to risk and just left it like this